pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1408,9 +1400,7 @@ self: super: {
 
   # The test suite depends on an impure cabal-install installation in
   # $HOME, which we don't have in our build sandbox.
-  cabal-install-parsers = dontCheck (super.cabal-install-parsers.override {
-    Cabal-syntax = self.Cabal-syntax_3_10_3_0;
```

Comment:
```
Warning:
    This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
      package cabal-install-parsers (cabal-install-parsers-0.6.1.1-8meeExLzuGzHHJO4ikZ4vI) requires Cabal-syntax-3.10.3.0-2U8Boie8mLdKbntrpHL5Qs
      package cabal-add (cabal-add-0.1) requires Cabal-syntax-3.10.3.0-2U8Boie8mLdKbntrpHL5Qs
      package cabal-add (cabal-add-0.1) requires Cabal-syntax-3.10.3.0-2U8Boie8mLdKbntrpHL5Qs
      package Cabal (Cabal-3.10.3.0-inplace) requires Cabal-syntax-3.10.3.0-inplace
(...)
*** abort because of serious configure-time warning from Cabal
```